xplique/wrappers/pytorch.py

```diff
@@ -0,0 +1,147 @@
+"""
+Module for having a wrapper for PyTorch's model
+"""
+try:
+    # use PyTorch functionality
+    import torch
+    from torch import nn
+    from torch import from_numpy
+except ImportError as exc:
+    raise ImportError(
+       "PyTorch is required to use this feature. Please install PyTorch using 'pip install torch'."
+    ) from exc
+
+import tensorflow as tf
+import numpy as np
+
+from ..types import Union, Optional, Tuple, Callable
+
+class TorchWrapper(tf.keras.Model):
+    """
+    A wrapper for PyTorch's model so that they can be used in Xplique framework
+    for most attributions method
+
+    Parameters
+    ----------
+    torch_model
+        A nn.Module PyTorch model
+    device
+        If we are on GPU or CPU
+    is_channel_first
+        A boolean that is true if the torch's model expect a channel dim and if this one come first
+    """
+
+    def __init__(self, torch_model: nn.Module, device: Union['torch.device', str],
+                 is_channel_first: Optional[bool] = None):
+        super().__init__()
+        self.model = torch_model.to(device)
+        self.device = device
+        # with torch, the convention for CNN is (N, C, H, W)
+        if is_channel_first is None:
+            self.channel_first = self._check_conv_layers()
+        else:
+            self.channel_first = is_channel_first
+        # deactivate all tf.function
+        tf.config.run_functions_eagerly(True)
+
+    # pylint: disable=arguments-differ
+    @tf.custom_gradient
+    def call(self, inputs: np.ndarray) -> Tuple[tf.Tensor, Callable]:
+        """
+        A method that allow to call the PyTorch model wrapped on inputs that are
+        numpy arrays reshaped to match Xplique's explainers expectations (i.e. channel-last)
+        , get the gradients from PyTorch framework and recast outputs and gradients
+        as Tensorflow tensors.
+
+        Parameters
+        ----------
+        inputs
+            Processed inputs as numpy arrays
+        
+        Returns
+        -------
+        outputs
+            Outputs of the PyTorch model converted to Tensorflow tensor
+        grad
+            The function that allow to compute the gradient of the PyTorch model and
+            broadcast it for Tensorflow
+        """
+        # transform your numpy inputs to torch
+        torch_inputs = self.transform_np_inputs(inputs).to(self.device)
+        torch_inputs.requires_grad_(True)
+
+        # make predictions
+        self.model.zero_grad()
+        outputs = self.model(torch_inputs)
+        output_tensor = tf.constant(outputs.cpu().detach().numpy())
+
+        def grad(upstream):
+            upstream_tensor = tf.constant(upstream.numpy())
+            torch.autograd.backward(
+                outputs.cpu(),
+                grad_tensors=from_numpy(upstream_tensor.numpy()),
+                retain_graph=True
```

Comment:
Good point, I guess at some points I needed it but the modification concerning the device seems to show that it is not necessary. As I don't remember exactly the purpose of it and since it seems to work without it I changed it